### PR TITLE
Move ScriptMsg to script_traits crate, Fixes #8833

### LINKED ISSUE #8833: Move ScriptMsg enum into script_traits crate
This message type is only sent between script threads and the constellation, which are respectively defined in the `script/` crate and the `compositing/` crate. We should be able to move the type into `script_traits/lib.rs`, which will allow for further simplification of our compilation graph in the future.

Code: `components/msg/constellation_msg.rs`, `components/script_traits/lib.rs`, `components/compositing/constellation.rs`

### PR DESCRIPTION
**Pull request by cnaj**
Review of documentation that was missing needed.

Fixes #8833.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8909)

### CODE REVIEW

File: components/script_traits/Cargo.toml [hunk 1]
```diff
@@ -7,18 +7,28 @@ authors = ["The Servo Project Developers"]
...
+
 [dependencies.plugins]
 path = "../plugins"
 
 [dependencies.profile_traits]
 path = "../profile_traits"
 
+[dependencies.style_traits]
+path = "../style_traits"
+
```
Comment (KiChjang) [inline components/script_traits/Cargo.toml]: Why were these 3 dependencies added?

Comment (cnaj) [inline components/script_traits/Cargo.toml]: `ScriptMsg` references `canvas_traits::CanvasMsg`, `offscreen_gl_context::GLContextAttributes` and `style_traits::viewport::ViewportConstraints`. Those dependencies were removed from `msg` crate.

**Comment by KiChjang** [comment]
This LGTM. Good call on making a separate `script_msg.rs` file.

r? @jdm

**Comment by jdm** [comment]
I agree! This just needs a `./mach cargo-update -p script_traits` to ensure that the other Cargo.lock files receive the appropriate updates automatically.

**Comment by KiChjang** [comment]
Looks like it's ready to merge. r? @jdm

**Comment by cnaj** [comment]
Thanks for supporting me on my first PR! @KiChjang, @jdm 
I'll sure look in to some more issues in the (near) future.

**Pull request merged**